### PR TITLE
Teacher Application: Fixed broken tests, fixed broken email params, m…

### PR DESCRIPTION
**Pull request by aoby**
…oved logic into the model (from controller), cleaned up model and added more validation & helpers.

CC @pcardune Thanks for getting the emails set up!

### CODE REVIEW

File: dashboard/app/models/pd/teacher_application.rb [hunk 1]
```diff
@@ -18,6 +18,33 @@
...
 class Pd::TeacherApplication < ActiveRecord::Base
+  PROGRAM_DETAILS_BY_COURSE = {
+    'csd' => {
+      name: 'CS Discoveries',
+      url: 'https://code.org/educate/professional-learning/cs-discoveries',
+      approval_form_id: '1FAIpQLSdcR6oK-JZCtJ7LR92MmNsRheZjODu_Qb-MVc97jEgxyPk24A'
+    },
+    'csp' => {
+      name: 'CS Principles',
+      url: 'https://code.org/educate/professinoal-learning/cs-principles',
```
Comment (mehalshah) [inline dashboard/app/models/pd/teacher_application.rb]: professional ?

File: dashboard/app/models/pd/teacher_application.rb [hunk 2]
```diff
@@ -18,6 +18,33 @@
...
+      approval_form_id: '1FAIpQLScVReYg18EYXvOFN2mQkDpDFgoVqKVv0bWOSE1LFSY34kyEHQ'
+    }
+  }
+
+  REQUIRED_APPLICATION_FIELDS = %w[
```
Comment (mehalshah) [inline dashboard/app/models/pd/teacher_application.rb:34]: There are a lot more required fields (survey, grades planning to teach, etc). Do you intend to validate all?

File: dashboard/app/models/pd/teacher_application.rb (hunk 1 above)
Comment (aoby) [reply dashboard/app/models/pd/teacher_application.rb]: oops, will fix

File: dashboard/app/models/pd/teacher_application.rb (hunk 2 above)
Comment (aoby) [reply dashboard/app/models/pd/teacher_application.rb:34]: yes. Do you have an exhaustive list?

**Pull request merged**